### PR TITLE
Add SingleStore connector

### PR DESCRIPTION
**Pull request by oyeliseiev-ua**
## Description
This PR introduces new JDBC connector for [SingleStore](https://www.singlestore.com/) .

## Motivation and Context

IBM lakehouse project requires new connector to integrate with SingleStore database.

## Impact

No impact 

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==
* Added SingleStore connector
```

### CODE REVIEW

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

File: presto-singlestore/src/test/java/com/facebook/presto/plugin/singlestore/SingleStoreQueryRunner.java
Comment (tdcmeehan) [inline presto-singlestore/src/test/java/com/facebook/presto/plugin/singlestore/SingleStoreQueryRunner.java:91]: Nit: please add a constant

**Review by tdcmeehan — COMMENTED** [review]
Overall this is a great submission, let me give it one more glance tomorrow/next week.  Thanks!

**Review by tdcmeehan — APPROVED** [review]

**Comment by tdcmeehan** [comment]
@oyeliseiev-ua please sign the CLA, then happy to merge.  Thanks!

**Comment by oyeliseiev-ua** [comment]
> @oyeliseiev-ua please sign the CLA, then happy to merge. Thanks!

singlestore-tests are not passed. Could you double check 'SINGLESTORE_LICENSE' secret value, for some reason test docker container is not started.

**Comment by evinaku** [comment]
@tdcmeehan @steveburnett Thank you for reviewing We are in the process of getting the CLA signed by legal, should be sorted in a couple of days.
Once thats done, it gets merged, will it be available to be used with 0.283 and just added to this [connectors](https://prestodb.io/docs/current/connector.html) list? or do we have to wait on the next release?

**Comment by tdcmeehan** [comment]
> @tdcmeehan @steveburnett Thank you for reviewing We are in the process of getting the CLA signed by legal, should be sorted in a couple of days. Once thats done, it gets merged, will it be available to be used with 0.283 and just added to this [connectors](https://prestodb.io/docs/current/connector.html) list? or do we have to wait on the next release?

It will be added in the next releases, which is slated for early October.

**Comment by tdcmeehan** [comment]
It appears tests failed after 20 minutes.  Should it be extended?

**Comment by oyeliseiev-ua** [comment]
> It appears tests failed after 20 minutes. Should it be extended?

yes, please. Thank you.

**Comment by tdcmeehan** [comment]
@oyeliseiev-ua you can make the tests run longer by updating the value here: https://github.com/prestodb/presto/pull/20781/files#diff-a566c4c55350532a4ec7e5b0149bb191937fe45fd5930f245c168560caea7ff8R37

**Comment by oyeliseiev-ua** [comment]
> @oyeliseiev-ua you can make the tests run longer by updating the value here: https://github.com/prestodb/presto/pull/20781/files#diff-a566c4c55350532a4ec7e5b0149bb191937fe45fd5930f245c168560caea7ff8R37

@tdcmeehan looks like some tests are failed with " no space left on device"

**Pull request merged**
